Userland/Libraries/LibCompress/PackBitsDecoder.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2023, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PackBitsDecoder.h"
+
+namespace Compress::PackBits {
+
+ErrorOr<ByteBuffer> decode_all(ReadonlyBytes bytes, Optional<u64> expected_output_size)
+{
+    auto memory_stream = make<FixedMemoryStream>(bytes);
+
+    ByteBuffer decoded_bytes;
+    TRY(decoded_bytes.try_ensure_capacity(expected_output_size.value_or(0)));
+
+    while (memory_stream->remaining() > 0 && decoded_bytes.size() <= expected_output_size.value_or(NumericLimits<u64>::max())) {
```

Comment:
Isn't encountering any of these before finding the end-of-stream symbol an error? Maybe less so for the second one, as that is just an optimization on our side so far, but the first one feels like it would silently accept corrupt data.

Reply:
Depends on who you are asking. As you also mentioned, the EOS symbol is not always present, so for some stream you just decode until there is nothing left (first condition). The second one is here because some ILBM decoders (and I suspect it to be done in other places) includes more data than necessary. So to stay compatible, we quit early if we have enough data.